userspace/libsinsp/sinsp_resource_utilization.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <cmath>
+#include "sinsp_resource_utilization.h"
+#include <sys/times.h>
+#include <sys/stat.h>
+#include "utils.h"
+#include <sinsp.h>
+#include "sinsp_public.h"
+#include <sys/sysinfo.h>
+
+
+struct sinsp_resource_utilization;
+
+void libsinsp::resource_utilization::get_rss_vsz_pss_memory(uint32_t &rss, uint32_t &vsz, uint32_t &pss)
+{
+	char filename[1024];
+	char line[512];
+	pid_t pid = getpid();
+
+	snprintf(filename, sizeof(filename), "/proc/%d/status", pid);
+	if(FILE* f = fopen(filename, "r"))
+	{
+		while(fgets(line, sizeof(line), f) != NULL)
+		{
+			if(strstr(line, "VmSize:") == line)
+			{
+				sscanf(line, "VmSize: %u", &vsz);		/* memory size returned in kb */
+				vsz /= 1024;		/* convert to MB with precision loss by design */
+			} else if(strstr(line, "VmRSS:") == line)
+			{
+				sscanf(line, "VmRSS: %u", &rss);		/* memory size returned in kb */
+				rss /= 1024;		/* convert to MB with precision loss by design */
+			}
+		}
+		fclose(f);
+	}
+
+    snprintf(filename, sizeof(filename), "/proc/%d/smaps_rollup", pid);
+	if(FILE* f = fopen(filename, "r"))
+	{
+		while(fgets(line, sizeof(line), f) != NULL)
+		{
+			if(strstr(line, "Pss:") == line)
```

Comment:
Same comment on using `strncmp`
```suggestion
			if(strncmp(line, "Pss:", 4) == 0)
```